src/script/script.h

```diff
@@ -571,16 +572,30 @@ class CScript : public CScriptBase
         if (b.empty())
             return nFound;
         iterator pc = begin();
+        iterator copied = begin();
+        size_t nShift = 0;
         opcodetype opcode;
         do
         {
             while (end() - pc >= (long)b.size() && memcmp(&pc[0], &b[0], b.size()) == 0)
             {
-                pc = erase(pc, pc + b.size());
+                // Lazy copy-in-place if there is a match:
+                if (nShift == 0) copied = pc;
```

Comment:
@kazcw :  The first time a match is found, copied will be begin(). nShift will be zero, and pc will be pointing into the array somewhere.

If we removed copied = pc and just did:
   ```copied = std::copy(copied+nShift, pc, copied);``

... then first is copied, last is pc, and result is copied.
copied IS in the range [copied, pc)